pkg/endpoint/endpoint.go

```diff
@@ -191,6 +191,11 @@ type Endpoint struct {
 	// Immutable after Endpoint creation.
 	containerIfName string
 
+	// parentIfIndex is the interface index of the ENI to which the endpoint is linked.
```

Comment:
Could we document this in a more generic manner? Is it reasonable that there might be a non-ENI use case?